pkg/scheduler/factory_test.go

```diff
@@ -273,145 +272,66 @@ func TestCreateFromConfigWithUnspecifiedPredicatesOrPriorities(t *testing.T) {
-		initFn                     func()
-		deferFn                    func()
 		injectErr                  error
 		podUpdatedDuringScheduling bool // pod is updated during a scheduling cycle
 		podDeletedDuringScheduling bool // pod is deleted during a scheduling cycle
-		expectNode                 *v1.Node
 		expect                     *v1.Pod
-		testBackoff                bool
 	}{
 		{
-			name:       "node is deleted during a scheduling cycle",
-			initFn:     func() { schedulerCache.AddNode(nodeFoo) },
-			injectErr:  apierrors.NewNotFound(apicore.Resource("node"), nodeFoo.Name),
-			expectNode: nodeFoo,
-		},
-		{
-			name: "pod is updated during a scheduling cycle",
-			initFn: func() {
-				queue.Add(testPod)
-				queue.Pop()
-			},
-			deferFn:                    func() { queue.Delete(testPod) },
+			name:                       "pod is updated during a scheduling cycle",
+			injectErr:                  nil,
 			podUpdatedDuringScheduling: true,
 			expect:                     testPodUpdated,
 		},
 		{
-			name: "pod is not updated during a scheduling cycle",
-			initFn: func() {
-				queue.Add(testPod)
-				queue.Pop()
-			},
-			deferFn: func() { queue.Delete(testPod) },
-			expect:  testPod,
-		},
-		{
-			name: "pod is not updated during a scheduling cycle",
-			initFn: func() {
-				queue.Add(testPod)
-				queue.Pop()
-			},
-			deferFn:     func() { queue.Delete(testPod) },
-			expect:      testPod,
-			testBackoff: true,
+			name:      "pod is not updated during a scheduling cycle",
+			injectErr: nil,
+			expect:    testPod,
 		},
 		{
-			name: "pod is deleted during a scheduling cycle",
-			initFn: func() {
-				queue.Add(testPod)
-				queue.Pop()
-			},
-			deferFn:                    func() { queue.Delete(testPod) },
+			name:                       "pod is deleted during a scheduling cycle",
+			injectErr:                  nil,
 			podDeletedDuringScheduling: true,
 			expect:                     nil,
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if tt.initFn != nil {
-				tt.initFn()
-			}
-			if tt.deferFn != nil {
-				defer tt.deferFn()
-			}
+			stopCh := make(chan struct{})
+			defer close(stopCh)
+
+			client := fake.NewSimpleClientset(&v1.PodList{Items: []v1.Pod{*testPod}})
+			informerFactory := informers.NewSharedInformerFactory(client, 0)
+			podInformer := informerFactory.Core().V1().Pods()
+			// Need to add/update/delete testPod to the store.
+			podInformer.Informer().GetStore().Add(testPod)
+
+			queue := internalqueue.NewPriorityQueue(nil, internalqueue.WithClock(clock.NewFakeClock(time.Now())))
+			schedulerCache := internalcache.New(30*time.Second, stopCh)
+
+			queue.Add(testPod)
+			queue.Pop()
 
-			// If pod is updated during the scheduling cycle.
 			if tt.podUpdatedDuringScheduling {
+				podInformer.Informer().GetStore().Update(testPodUpdated)
 				queue.Update(testPod, testPodUpdated)
 			} else if tt.podDeletedDuringScheduling {
```

Comment:
nit: remove the else. In case we want to add a test of subsequent update->delete in the future.

Reply:
done.